src/jquery.uls.lcd.js

```diff
@@ -55,17 +55,101 @@
 
 		this.$element.addClass( 'uls-lcd' );
 		this.regionLanguages = {};
+		this.$languageOptionListItems = null;
 		this.renderTimeout = null;
 		this.$cachedQuicklist = null;
 		this.groupByRegionOverride = null;
 
+		// The index of the language option that is currently visited using arrow key navigation
+		// Can take values in the [0, languageOptionListItemsLength - 1] range for top to bottom
+		// navigation, or in the [-1, -languageOptionListItemsLength + 1] range for bottom to top
+		// navigation.
+		this.navigationIndex = null;
+
 		this.render();
 		this.listen();
 	}
 
 	LanguageCategoryDisplay.prototype = {
 		constructor: LanguageCategoryDisplay,
 
+		/**
+		 * Returns a jQuery object containing a collection of all the
+		 * language option <li> elements
+		 *
+		 * @return {jQuery}
+		 */
+		getLanguageOptionListItems: function () {
```

Comment:
We might be able to cache the result of this method in a function, but it may not be safe to do it on a "class" level since if any other modifications are done, it would not reflect in the property.

For example if an item is added it would not be reflected in this property.